spec/controllers/your_details_controller_spec.rb

```diff
@@ -54,33 +65,24 @@ module Wpcc
       context 'success' do
         after { Timecop.return }
 
-        let(:your_details_form) { Wpcc::YourDetailsForm.new(nil_session) }
-        let(:nil_session) do
-          {
-            'age' => nil,
-            'gender' => nil,
-            'salary' => nil,
-            'salary_frequency' => nil,
-            'contribution_preference' => nil
-          }
-        end
+        let(:your_details_form) { Wpcc::YourDetailsForm.new(some_nil_session) }
 
         it 'stores the form input in a session for 30 minutes' do
```

Comment:
I agree that it is testing 2 things. I struggled to create tests that conveyed and proved the time aspect of the session expiring. This test was meant to show that it stores details and after the session expires the session is updated - so it compares how the session details change after the time limit. I'll make the suggested change and see what you think. Perhaps the tests below convey my original intention?